core/templates/components/forms/schema-based-editors/schema-based-editor.component.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for general schema-based editors.
+ */
+
+import { Input, Output, EventEmitter, Component, forwardRef, AfterViewInit, ViewChild, ElementRef } from '@angular/core';
+import { NG_VALUE_ACCESSOR, NG_VALIDATORS, ControlValueAccessor, Validator, AbstractControl, ValidationErrors, NgForm } from '@angular/forms';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { Schema } from 'services/schema-default-value.service';
+const INVALID = 'INVALID';
```

Comment:
The name of this should include a description of its "category".

Reply:
Changed to `VALIDATION_STATUS_INVALID`.